### ISSUE TITLE
updatewcs parameter 'use_db' default value=True causes crash for most users

### ISSUE THREAD

**Issue by mackjenn** [spacetelescope/stwcs#70]
We keep getting help tickets describing errors when running updatewcs.

>updatewcs.updatewcs('*flc.fits')
TypeError: Can't convert 'NoneType' object to str implicitly

To solve this, we are telling users to set a new parameter which is not documented in any of our drizzling examples:
>updatewcs.updatewcs('*flc.fits', use_db=False)

Can we simply change the default to False so that people trying to reprocess their data don't keep tripping over this?

Mihai pointed me to some open isseus:
https://github.com/spacetelescope/stwcs/issues/38
https://github.com/spacetelescope/stwcs/pull/40

He suggests: 'Maybe the code should be fixed to automatically ignore astrometry_db errors and proceed as if 'use_db=False' ?

**Comment by mcara** [comment]
After looking at the code (which I do not understand completely) it looks like it is a series of bugs. First of all, [`serviceEndPoint`](https://github.com/spacetelescope/stwcs/blob/a54afef6050161cffd4e91b70dbe75c77b7453c2/stwcs/updatewcs/astrometry_utils.py#L327) seems to be constructed in a not very stable way (hence the code is crashing when one of the variables is `None` but then the headerlet is not constructed well (FITS check fails).

**Comment by mcara** [comment]
FYI, the reported error is:
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-2-870b0bd17ccc> in <module>()
----> 1 updatewcs('ichx03iaq_flt.fits')

~/.../python3.5/site-packages/stwcs/updatewcs/__init__.py in updatewcs(input, vacorr, tddcorr, npolcorr, d2imcorr, checkfiles, verbose, use_db)
    120             # Add any new astrometry solutions available from
    121             #  an accessible astrometry web-service
--> 122             astrometry.updateObs(f)
    123 
    124     return files

~/.../python3.5/site-packages/stwcs/updatewcs/astrometry_utils.py in updateObs(self, obsname)
    165         hdrnames = headerlet.get_headerlet_kw_names(fileobj, 'hdrname')
    166 
--> 167         headerlets, best_solution_id = self.getObservation(observationID)
    168         if headerlets is None:
    169             logger.warning("Problems getting solutions from database")

~/.../python3.5/site-packages/stwcs/updatewcs/astrometry_utils.py in getObservation(self, observationID)
    325                 serviceEndPoint = self.serviceLocation + \
    326                     'observation/read/' + observationID + \
--> 327                     '?wcsname='+solutionID
    328                 r_solution = requests.get(serviceEndPoint, headers=headers)
    329                 if r_solution.status_code == requests.codes.ok:

TypeError: Can't convert 'NoneType' object to str implicitly
```

**Comment by mcara** [comment]
@mackjenn A fix was proposed in #71

**Comment by mcara** [comment]
Actually the fix was already implemented in https://github.com/spacetelescope/stwcs/pull/62. #71 simply improves the code a little (mostly aesthetically) given #62.

The main issue is that these changes have not yet been released publicly. Just tagging the repository is not enough:

> 2018-08-07 13:31:05 -0400  (tag: 1.4.2, upstream/stable)

One must also edit astroconda and increase the version of the package to match the tag.

CC: @stsci-hack @nden @rendinam

**Comment by nden** [comment]
fixed in #62 and #71
